### PR TITLE
Display mark for remote branches with tracking branches

### LINKED ISSUE #847: mark remote branches having tracking branches
right now we only mark the local branches that are tied to a remote tracking branch:

Screenshot 2021-08-15 at 18 23 51

would be nice to have a similar marker for the corresponding remote branches:

Screenshot 2021-08-15 at 18 24 21

### PR DESCRIPTION
**Pull request by jedel1043**
Fixes #847.

First contribution here! I'm still getting used to the codebase so please tell me if I overlooked a simpler solution.
Anyways, I choosed the U+2193 (↓) character to represent a remote branch with a tracking branch; it matches with the U+2190 (↑) symbol that represents a local branch with a remote tracking branch. If you find a more fitting symbol for this, I'll gladly change it.

### CODE REVIEW

**Comment by extrawurst** [comment]
Hi @jedel1043 thanks for tackling this. looks pretty good already. can you please add a test for `get_branches_info` that specifically tests for the new `has_tracking` correctly set?

**Comment by jedel1043** [comment]
> Hi @jedel1043 thanks for tackling this. looks pretty good already. can you please add a test for `get_branches_info` that specifically tests for the new `has_tracking` correctly set?

I'm on it.

**Comment by jedel1043** [comment]
Added test and the auxiliar function `remote_details` to complement the function `local_details`. Let me know if there's something else missing.

File: asyncgit/src/sync/branch/mod.rs [hunk 1]
```diff
@@ -88,6 +97,14 @@ impl BranchInfo {
 
 		None
 	}
+	/// returns details about remote branch or None
+	pub const fn remote_details(&self) -> Option<&RemoteBranch> {
```
Comment (extrawurst) [inline asyncgit/src/sync/branch/mod.rs:101]: since we only ever use it inside the tests, lets move it there

**Review by extrawurst — CHANGES_REQUESTED** [review]
👍 just one last comment

File: asyncgit/src/sync/branch/mod.rs (hunk 1 above)
Comment (jedel1043) [reply asyncgit/src/sync/branch/mod.rs]: Solved.

**Comment by extrawurst** [comment]
@jedel1043 thanks for your contribution ❤️

**Comment by jedel1043** [comment]
> @jedel1043 thanks for your contribution ❤️

Thank you for the quick reviews! 😁
I hope I can contribute a lot more to this project!

**Pull request merged**